src/main/java/dev/openfeature/sdk/ImmutableContext.java

```diff
@@ -0,0 +1,123 @@
+package dev.openfeature.sdk;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Delegate;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The EvaluationContext is a container for arbitrary contextual data
+ * that can be used as a basis for dynamic evaluation.
+ * The MutableContext is an EvaluationContext implementation which is not threadsafe, and whose attributes can
+ * be modified after instantiation.
+ */
+@ToString
+@SuppressWarnings("PMD.BeanMembersShouldSerialize")
+public class ImmutableContext implements EvaluationContext {
```

Comment:
ImmutableContext must be `final`.

You will need to change constructors to support this requirement.